pkg/cloudprovider/providers/azure/azure_backoff.go

```diff
@@ -142,58 +96,43 @@ func (az *Cloud) CreateOrUpdateLBWithRetry(lb network.LoadBalancer) error {
 		resp := <-respChan
 		err := <-errChan
 		glog.V(10).Infof("LoadBalancerClient.CreateOrUpdate(%s): end", *lb.Name)
-		return processRetryResponse(resp.Response, err)
+		done, err := processRetryResponse(resp.Response, err)
+		if done && err == nil {
+			// Update the cache right after updating.
+			az.lbCache.AddOrUpdate(*lb.Name, nil)
```

Comment:
shouldn't the 2nd arg be the (*LB)

Reply:
This is one reason why existing cache not fit for us: after virtual machine updates, we want an updated struct from Azure API even if it is not expired yet (so that the next Get call will return the new structure).

Reply:
The next Get call should add the key then. nil values are dangerous here. If i am writing against the interface i should expect Get to return either `the struct instance` for cached or nil for `not cached`.